addons/jest/src/components/Message.tsx

```diff
@@ -0,0 +1,208 @@
+    {msg
+      .filter(item => typeof item !== 'string' || item.trim() !== '')
+      .map((item, index, list) => {
+        if (typeof item === 'string') {
+          if (index === 0 && index === list.length - 1) {
+            return item.trim();
+          }
+          if (index === 0) {
+            return item.replace(/^[\s\n]*/, '');
+          }
+          if (index === list.length - 1) {
+            return item.replace(/[\s\n]*$/, '');
+          }
+        }
+        return item;
+      })}
+  </section>
+))({
+  padding: 10,
+});
+
+interface SubgroupOptions {
+  startTrigger: (e: MsgElement) => boolean;
+  endTrigger: (e: MsgElement) => boolean;
+  grouper: (list: MsgElement[], key: number) => JSX.Element;
+  accList?: MsgElement[];
+  grouped?: MsgElement[];
+  grouperIndex?: number;
+  mode?: 'inject' | 'stop';
+  injectionPoint?: number;
+}
+
+const createSubgroup = ({
+  startTrigger,
+  endTrigger,
+  grouper,
+  accList = [],
+  grouped = [],
+  grouperIndex = 0,
+  mode,
+  injectionPoint,
+}: SubgroupOptions) => (acc: MsgElement[], item: MsgElement, i: number, list: MsgElement[]) => {
+  grouperIndex += 1;
+
+  // start or stop extraction
+  if (startTrigger(item)) {
+    mode = 'inject';
+    injectionPoint = i;
+  }
+  if (endTrigger(item)) {
+    mode = 'stop';
+  }
+
+  // push item in correct aggregator
+  if (mode === 'inject') {
+    grouped.push(item);
+  } else {
+    accList.push(item);
+  }
+
+  // on last iteration inject at detected injection point, and group
+  if (i === list.length - 1) {
+    // Provide a "safety net" when Jest returns a partially recognized "group"
+    // (recognized by acc.startTrigger but acc.endTrigger was never found) and
+    // it's the only group in output for a test result. In that case, accList
+    // will be empty, so return whatever was found, even if it will be unstyled
+    // and prevent next createSubgroup calls from throwing due to empty lists.
+    accList.push(null);
+
+    return accList.reduce<MsgElement[]>((eacc, el, ei) => {
+      switch (true) {
+        case injectionPoint === 0 && ei === 0: {
+          // at index 0, inject before
+          return eacc.concat(grouper(grouped, grouperIndex)).concat(el);
+        }
+        case injectionPoint > 0 && injectionPoint === ei + 1: {
+          // at index > 0, and next index WOULD BE injectionPoint, inject after
+          return eacc.concat(el).concat(grouper(grouped, grouperIndex));
+        }
+        default: {
+          // do not inject
+          return eacc.concat(el);
+        }
+      }
+    }, []);
+  }
+  return acc;
+};
+
+interface MessageProps {
+  msg: string;
+}
+
+type MsgElement = string | JSX.Element;
+
+const Message = ({ msg }: MessageProps) => {
+  const data = patterns
+    .reduce((acc, regex) => acc.replace(regex, ''), msg)
+    .split(/\[2m/)
+    .join('')
+    .split(/\[22m/)
+    .reduce<string[]>((acc, item) => acc.concat(item), [])
+    .map((item, li) =>
+      typeof item === 'string'
```

Comment:
are these type checks necassary? given the above reduce you should be getting an array of strings